pyanaconda/modules/storage/partitioning/interactive/change_device.py

```diff
@@ -0,0 +1,268 @@
+        log.debug("Modifying an existing device.")
+
+        self._revert_device_reformat()
+        self._change_device_size()
+
+        if self._should_reformat_device():
+            self._change_device_encryption()
+            self._change_device_format()
+        else:
+            self._change_device_label()
+            self._change_device_mount_point()
+
+        self._change_device_name()
+
+    def _revert_device_reformat(self):
+        """Revert reformat of the device."""
+        if self._request.reformat:
+            return
+
+        log.debug("Reverting device reformat.")
+        revert_reformat(self._storage, self._device)
+
+    def _change_device_size(self):
+        """Resize the device."""
+        size = Size(self._request.device_size)
+        original_size = Size(self._original_request.device_size)
+
+        if size == original_size:
+            return
+
+        log.debug("Changing device size: %s", size)
+        resize_device(self._storage, self._device, size, original_size)
+
+    def _should_reformat_device(self):
+        """Should we reformat the device?
+
+        It's possible that reformat is requested but the format
+        type is unchanged, in which case we're not going to schedule
+        another reformat unless encryption has changed.
+
+        :return: True of False
+        """
+        if not self._request.reformat:
+            return False
+
+        if self._device.format.exists:
+            return True
+
+        if self._original_request.device_encrypted != self._request.device_encrypted:
+            return True
+
+        if self._original_request.luks_version != self._request.luks_version:
+            return True
+
+        if self._original_request.format_type != self._request.format_type:
+            return True
+
+        return False
+
+    def _change_device_encryption(self):
+        """Change the device encryption."""
+        storage = self._storage
+        device = self._device
+        encrypted = self._request.device_encrypted
+        luks_version = self._request.luks_version
+
+        if self._original_request.device_encrypted != encrypted:
+            log.debug("Changing device encryption: %s", encrypted)
+            device = change_encryption(
+                storage=storage,
+                device=device,
+                encrypted=encrypted,
+                luks_version=luks_version
+            )
+        elif encrypted and self._original_request.luks_version != luks_version:
+            log.debug("Changing LUKS version: %s", luks_version)
+            device = change_encryption(
+                storage=storage,
+                device=device,
+                encrypted=False,
+                luks_version=luks_version
+            )
+
+            device = change_encryption(
+                storage=storage,
+                device=device,
+                encrypted=True,
+                luks_version=luks_version
+            )
```

Comment:
Why you have two calls here. One without encryption and one with encryption? Could you please add comment explaining that? Or abstract these calls somewhere to make the reason more obvious.

Reply:
Fixed.